src/MDMTransaction.h

```diff
@@ -27,9 +27,36 @@ NS_SWIFT_NAME(Transaction)
 
 #pragma mark Adding plans to a transaction
 
-/** Associate an plan with a given target. */
+/**
+Associates a plan with a given target.
+
+@param plan The plan to add to this transaction.
+@param target The target on which the plan can operate.
+*/
+- (void)addPlan:(nonnull id<MDMPlan>)plan
+       toTarget:(nonnull id)target
+  NS_SWIFT_NAME(add(plan:to:));
+
+/**
+ Associates a named plan with a given target.
+ 
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+ @param name String identifier for the plan.
+ */
 - (void)addPlan:(nonnull id<MDMPlan>)plan
        toTarget:(nonnull id)target
-    NS_SWIFT_NAME(add(plan:to:));
+       withName:(nonnull NSString *)name
+  NS_SWIFT_NAME(add(plan:to:withName:));
+
+/**
+ Removes a named plan from the transaction.
+ 
+ @param plan The plan to add to this transaction.
+ @param target The target on which the plan can operate.
+ @param name String identifier for the plan
+ */
+- (void)removePlanNamed:(nonnull NSString *)name
+  NS_SWIFT_NAME(remove(name:));
```

Comment:
I can see the reasoning that led to this change in the api; we may want to revisit the starmap transaction api to make its desired behavior more clear.

Aside:
Specifically, each "operation" made to a transaction is meant to be eventually committed to the scheduler. A cleaner api for this behavior may have been `append(operation:)` where operation is one of AddPlanOperation, AddNamedPlanOperation, and RemoveNamedPlanOperation, with each of these operations being transferrable to the scheduler so that it can take action on them.